networking/external_dns_operator/nw-external-dns-operator-configuration-parameters.adoc

```diff
@@ -0,0 +1,90 @@
+
+[source,yaml]
+----
+spec:
+  provider:
+    type: AWS <1>
+    aws:
+      credentials:
+        name: aws-access-key <2>
+----
+<1> `type` field defines available options such as AWS, GCP and Azure.
+<2> `name` field defines a name of the `secret` which contains credentials for your cloud provider.
+
+|`zones`
+|`zones` enables your specified AWS zone. When you do not specify zones, `ExternalDNS` discovers all the zones present in your cloud provider account.
+
+[source,yaml]
+----
+zones:
+- "myzoneid" <1>
+----
+
+<1> The `zones` specifies the IDs of hosted zones.
+
+|`domains`
+|`domains` field defines the hostnames that the External DNS Operator considers for the record creation to the zone specified in zones section.
+
+[source,yaml]
+----
+domains:
+- filterType: Include <1>
+  matchType: Exact <2>
+  name: "myzonedomain1.com" <3>
+- filterType: Include
+  matchType: Pattern <4>
+  pattern: ".*\\.otherzonedomain.com" <5>
+----
+<1> The `Include` field checks if the domain is present in the hostname.
+<2> If the source is route then the External DNS Operator checks the `domain` of route hostname. For Example, If the hostname is `test.myzonedomain1.com` then External DNS Operator filters the hostname to create the DNS records in the defined zone ID's present under zone's section.
+<3> The `name` defines the domain name that External DNS Operator filters.
+<4> The `matchType` sets `regex-domain-filter` in external DNS. You can limit possible domains and target zones by a Regex filter.
+<5> The `pattern` defines the regex pattern to be used by External DNS Operator to filter domains and target zones.
+
+|`source`
+|`source`
+
+[source,yaml]
+----
+source: <1>
+  type: Service <2>
+  service:
+    serviceType:<3>
+      - LoadBalancer
+      - ClusterIP
+  labelFilter: <4>
+    matchLabels:
+      external-dns.mydomain.org/publish: "yes"
+  hostnameAnnotation: "Allow" <5>
+  fqdnTemplate:
+  - "{{.Name}}.myzonedomain.com" <6>
+----
+<1> The `source` fields defines the resource types.
+<2> External DNS filters source type of `service` for creating dns records.
+<3> The `serviceType` sets `service-type-filter` in external-dns. The `serviceType` contains `default`: `all`, `expected`: `ClusterIP`, `NodePort`, `LoadBalancer` or `ExternalName` fields.
+<4> The `labelFilter` ensures that the controller considers only those resources which matches with label filter.
+<5> The default value for `hostnameAnnotation` is `Ignore`. When the value is `Allow` the DNS records get generated based on the value specified in the `external-dns.alpha.kubernetes.io/hostname`. External DNS Operator generates DNS records using the templates specified in the field `fqdnTemplates`.
+<6> External DNS Operator uses a string to generate DNS names from sources that don't define a hostname, or to add a hostname suffix when paired with the fake source.
+
+[source,yaml]
+----
+source:
+  type: OpenShiftRoute <1>
+  openshiftRouteOptions:
+    routerName: default <2>
```

Comment:
I think `labelFilter` below is also supported for routes. @miheer Do you think we should add it?
```
  labelFilter:
    matchLabels:
      external-dns.mydomain.org/publish: "yes"
```

Reply:
@miheer can you please confirm?

Reply:
I will add the fields